src/Illuminate/Foundation/Testing/TestCase.php

```diff
@@ -197,6 +197,11 @@ protected function tearDown(): void
 
             ParallelTesting::callTearDownTestCaseCallbacks($this);
 
+            $database = $this->app['db'];
```

Comment:
It is possible to run an application without `db`

Reply:
Ah.. Exactly!
Fixed in https://github.com/laravel/framework/pull/49327/commits/8dc1054f3e907b019ef26b1ee288d1d40d94ae96.